File: python/encapsulamiento/tarjeta_baja/tarjeta_baja.py
"""class TarjetaBaja"""


class TarjetaBaja():

    def __init__(self, saldo_inicial):
        self.__saldo = saldo_inicial
        self.__viajes_en_colectivo = 0
        self.__viajes_en_subte = 0

    def obtener_saldo(self):
        return self.__saldo

    def pagar_viaje_en_colectivo(self):
        if self.__hay_saldo_para_un_bondi():
            self.__saldo -= 21.50
            self.__viajes_en_colectivo += 1

    def pagar_viaje_en_subte(self):
        if self.__hay_saldo_para_un_subte():
            self.__saldo -= 19.50
            self.__viajes_en_subte += 1

    def contar_viajes(self):
        return self.__viajes_en_colectivo + self.__viajes_en_subte

    def contar_viajes_en_colectivo(self):
        return self.__viajes_en_colectivo

    def contar_viajes_en_subte(self):
        return self.__viajes_en_subte

    def __hay_saldo_para_un_bondi(self):
        return self.__saldo >= 21.50

    def __hay_saldo_para_un_subte(self):
        return self.__saldo >= 19.50

File: java/ENC/EncTN/src/TarjetaBaja.java
public class TarjetaBaja {

    private double saldo;
    private int viajesEnColectivo;
    private int viajesEnSubte;

    public static double VIAJE_EN_SUBTE = 19.50;
    public static double VIAJE_EN_COLECTIVO = 21.50;

    /**
     * post: saldo de la Tarjeta en saldoInicial.
     */
    public TarjetaBaja(double saldoInicial) {
        saldo = saldoInicial;
        viajesEnColectivo = 0;
        viajesEnSubte = 0;

    }

    /**
     * post: devuelve el saldo actual de la Tarjeta
     */
    public double obtenerSaldo() {
        return saldo;
    }

    /**
     * post: agrega el monto al saldo de la Tarjeta.
     */
    public void cargar(double monto) {
        saldo += monto;
    }

    /**
     * pre : saldo suficiente.
     * post: utiliza 21.50 del saldo para un viaje en colectivo.
     */
    public void pagarViajeEnColectivo() {
        if (saldo >= VIAJE_EN_COLECTIVO) {
            saldo -= VIAJE_EN_COLECTIVO;
            viajesEnColectivo++;
        }
    }

    /**
     * pre : saldo suficiente.
     * post: utiliza 19.50 del saldo para un viaje en subte.
     */
    public void pagarViajeEnSubte() {
        if (saldo >= VIAJE_EN_SUBTE) {
            saldo -= VIAJE_EN_SUBTE;
            viajesEnSubte++;
        }
    }

    /**
     * post: devuelve la cantidad de viajes realizados.
     */
    public int contarViajes() {

        return viajesEnColectivo + viajesEnSubte;
    }

    /**
     * post: devuelve la cantidad de viajes en colectivo.
     */
    public int contarViajesEnColectivo() {

        return viajesEnColectivo;
    }

    /**
     * post: devuelve la cantidad de viajes en subte.
     */
    public int contarViajesEnSubte() {
        return viajesEnSubte;
    }
}